contracts/FlowIDTableStaking.cdc

```diff
@@ -993,7 +990,31 @@ pub contract FlowIDTableStaking {
         }
     }
 
-    /// Functions to return contract fields
+    pub fun getNetworkingAddressClaimed(address: String): Bool {
+        return self.getClaimed(path: /storage/networkingAddressesClaimed, key: address)
+    }
+
+    pub fun getNetworkingKeyClaimed(key: String): Bool {
+        return self.getClaimed(path: /storage/networkingKeysClaimed, key: key)
+    }
+
+    pub fun getStakingKeyClaimed(key: String): Bool {
+        return self.getClaimed(path: /storage/stakingKeysClaimed, key: key)
+    }
+
+    access(account) fun getClaimed(path: StoragePath, key: String): Bool {
+        let claimedDictionary = self.account.load<{String: Bool}>(from: path)
+            ?? panic("Invalid path for dictionary")
+
+        var claimed = false
+        if claimedDictionary[key] != nil {
+            claimed = true
+        } else {
+            claimed = false
+        }
+        self.account.save(claimedDictionary, to: path)
+        return claimed
```

Comment:
No need to `load` and re-`save` after the read, you can either just `copy` or `borrow`, and then shorten the if-statement to nil-coaelescing to provide a default if there is no entry:

```suggestion
		let claimedDictionary = self.account.borrow<&{String: Bool}>(from: path)
            ?? panic("Invalid path for dictionary")
        return claimedDictionary[key] ?? false
```

Reply:
Good suggestion!